### PR TITLE
[V0094, V0095 - merge after PR #3223] Change /totals/by_entity/ to pull from entity_chart_vw; create ofec_agg_coverage_date_mv;

### PR DESCRIPTION
**Pull request by hcaofec**
## Summary 

- Resolves #[3240](https://github.com/fecgov/openFEC/issues/3240)

_entity_chart_vw is created, and /totals/by_entity/ endpoint is using this view now._

## How to test the changes locally

-Run this branch locally and try endpoint /totals/by_entity/ ( using dev database )

## Impacted areas of the application
List general components of the application that this PR will affect:

-  raising and spending chart

### CODE REVIEW

File: data/migrations/V0089__add_ofec_agg_coverage_date_mv.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  SELECT row_number() OVER () AS idx,
+         f.cmte_id AS cand_cmte_id,
+         f.rpt_yr + f.rpt_yr % 2::numeric AS fec_election_yr,
+         max(f.cvg_end_dt) AS cvg_end_date
+   FROM   disclosure.v_sum_and_det_sum_report f
+  WHERE f.orig_sub_id IN ( SELECT o.sub_id
+                             FROM staging.operations_log o
+                            WHERE o.pass_3_entry_done_dt IS NOT NULL) 
+    AND f.form_tp_cd::text ~~ 'F3%'::text
```
[the remark this answers is not in the document]
Comment (hcaofec) [reply data/migrations/V0089__add_ofec_agg_coverage_date_mv.sql]: I don't think there is a big difference in this particular case. Unless a function-based index on left() is created to use the "left()",  or a gin index with gin_trgm_ops to use the "like",  they both do full table scan.  Here is the result by analyze two queries:
like 'F3%' : 
Planning time: 1.230 ms
Execution time: 4981.833 ms
(15 rows)

left()='F3'
Planning time: 0.351 ms
Execution time: 7389.459 ms
(19 rows)

**Comment by fecjjeng** [comment]
Refresh MV public.ofec_agg_coverage_date_mv failed:
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) cannot refresh materialized view "public.ofec_agg_coverage_date_mv" concurrently
HINT:  Create a unique index with no WHERE clause on one or more columns of the materialized view.
 [SQL: 'REFRESH MATERIALIZED VIEW CONCURRENTLY ofec_agg_coverage_date_mv']

**Comment by fecjjeng** [comment]
The mv got refreshed now without error.  Good job.

**Comment by fecjjeng** [comment]
However, this mv does not have data in the sample db.  Joseph had put in 277 rows in disclosure.v_sum_and_det_sum_report and 215 rows in staging.operations_log.  However, data in these two does not match this new view's criteria so this new mv has no data.  You may need to add some more rows in either one of these two tables to generate some rows for this new mv.

**Comment by hcaofec** [comment]
yes, new sample data need s to be added otherwise the mv has no data.

File: data/migrations/V0089__add_ofec_agg_coverage_date_mv.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (lbeaufort) [reply data/migrations/V0089__add_ofec_agg_coverage_date_mv.sql]: OK thanks @hcaofec!

**Comment by fecjjeng** [comment]
The latest change had been downloaded to local.  sample_db had been created locally and ofec_agg_coverage_date_mv had been successfully refreshed and has data in it.  Great job!

**Pull request merged**